daemon/image_pull.go

```diff
@@ -104,3 +104,39 @@ func (daemon *Daemon) pullImageWithReference(ctx context.Context, ref reference.
 	<-writesDone
 	return err
 }
+
+func (daemon *Daemon) ResolveTagToDigest(ctx context.Context, ref reference.NamedTagged, authConfig *types.AuthConfig) (digest.Digest, error) {
```

Comment:
Will make this function return a fully configured repository to expand its usability.